aws/aws-how-to/instances/launch-ubuntu-desktop.rst

```diff
@@ -1,78 +1,105 @@
 
-To install Ubuntu desktop and TightVNC server on your EC2 instance, SSH into it and run:
+Create a security group
+-----------------------
+
+Select :guilabel:`Create Security Group`. Make sure the ssh tick box is selected to allow remote access.
+
+You can select an existing key pair if you already have one set up or create a new key pair. To create a key pair give it a name and Select ``RSA`` and ``PEM``. A Pem file will be automatically downloaded.
+
+Launch the instance and connect to it either via ssh or the AWS Console.
+
+Install Ubuntu Desktop Packages
+-------------------------------
+
+.. code::
+
+    sudo apt update && apt upgrade -y
+    sudo apt install -y ubuntu-desktop
+    sudo snap install snap-store --edge
+
+Install and Configure RDP
+-------------------------
+
+Install the xrdp server.
 
 .. code::
 
-    sudo apt update
-    sudo apt install ubuntu-desktop
-    sudo apt install tightvncserver
-    sudo apt install gnome-panel gnome-settings-daemon metacity nautilus gnome-terminal
+    sudo apt install -y xrdp
 
-Save the password created during the installation of the VNC server.
+Configure xrdp to use SSL to get an encrypted connection.
 
+.. code::
 
-Configure the VNC server
-------------------------
+    sudo usermod -a -G ssl-cert xrdp
 
-On your VM, launch the VNC server to create an initial configuration file:
+Set up a password for the Ubuntu user.
 
 .. code::
 
-    vncserver :1
+    passwd
 
-Edit the configuration file ``~/.vnc/xstartup`` to include:
+Finally, restart the xrdp service.
 
 .. code::
 
-    #!/bin/sh
+    systemctl restart xrdp
+
+Configure the Ubuntu Session
+----------------------------
 
-    export XKL_XMODMAP_DISABLE=1
-    unset SESSION_MANAGER
-    unset DBUS_SESSION_BUS_ADDRESS
+Connect to your instance using RDP to check the previous steps were succseful. Create a configuration script called ``ubuntu-session`` in ``/usr/local/bin/`` to run on RDP connections.
 
-    [ -x /etc/vnc/xstartup ] && exec /etc/vnc/xstartup
-    [ -r $HOME/.Xresources ] && xrdb $HOME/.Xresources
-    xsetroot -solid grey
+.. code::
+
+    sudo nano /usr/local/bin/ubuntu-session
+
+Add the following to the ubuntnu-session script.
 
-    vncconfig -iconic &
-    gnome-panel &
-    gnome-settings-daemon &
-    metacity &
-    nautilus &
-    gnome-terminal &
+.. code::
+
+    #!/bin/sh
 
+    export GNOME_SHELL_SESSION_MODE=ubuntu
+    export DESKTOP_SESSION=ubuntu-xorg
+    export XDG_SESSION_DESKTOP=ubuntu-xorg
+    export XDG_CURRENT_DESKTOP=ubuntu:GNOME
+    
+    exec /usr/bin/gnome-session --session=ubuntu
 
-Kill and restart the VNC server:
+Make the script executable.
 
 .. code::
 
-    vncserver -kill :1
+    sudo chmod +x /usr/local/bin/ubuntu-session
 
-    vncserver :1
+Update the session manager to use the new session configuration.
 
+.. code::
+
+    update-alternatives --install /usr/bin/x-session-manager x-session-manager /usr/local/bin/ubuntu-session 60
 
 Allow traffic on the VNC port
```

Comment:
For this and the following sections: Since we have set up the RDP server, and plan to use that, these sections and references to VNC are not needed. Just use a single consistent approach as mentioned in the discourse post.